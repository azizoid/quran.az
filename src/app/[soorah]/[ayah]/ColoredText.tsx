export const coloredTextColors = [
  '#3490dc',
  '#6574cd',
  '#9561e2',
  '#f66d9b',
  '#f6993f',
  '#38c172',
  '#4dc0b5',
  '#6cb2eb',
]

type ColoredTextProps = {
  content: string
}

export const ColoredText = ({ content }: ColoredTextProps) => {
  const text = content.split(' ')

  return (
    <div className="flex flex-wrap">
      {text.map((word, index) => (
        <span
          key={index}
          style={{ color: coloredTextColors[index % coloredTextColors.length] }}
          className="px-1 py-0.5 hover:bg-gray-100 hover:cursor-pointer"
        >
          {word}{' '}
        </span>
      ))}
    </div>
  )
}
