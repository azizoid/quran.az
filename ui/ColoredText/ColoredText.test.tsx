import React from 'react'
import {render} from '@testing-library/react'

import { ColoredText, coloredTextcolors } from './ColoredText'

test("render Component", ()=>{
  const { container, getByText} = render(<ColoredText content='Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed' />)

  expect(getByText('Lorem')).toHaveStyle(`color:${coloredTextcolors[0]}`)
  expect(getByText('amet,')).toHaveStyle(`color:${coloredTextcolors[4]}`)
  expect(getByText('consectetur')).toHaveStyle(`color:${coloredTextcolors[5]}`)
  expect(getByText('sed')).toHaveStyle(`color:${coloredTextcolors[0]}`)

  expect(container).toMatchSnapshot()
})