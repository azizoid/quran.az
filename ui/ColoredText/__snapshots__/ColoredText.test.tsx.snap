// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Component 1`] = `
<div>
  <div
    class="flex flex-wrap"
  >
    <span
      class="px-1 py-0.5 hover:bg-gray-100 hover:cursor-pointer"
      style="color: rgb(52, 144, 220);"
    >
      Lorem 
    </span>
    <span
      class="px-1 py-0.5 hover:bg-gray-100 hover:cursor-pointer"
      style="color: rgb(101, 116, 205);"
    >
      ipsum 
    </span>
    <span
      class="px-1 py-0.5 hover:bg-gray-100 hover:cursor-pointer"
      style="color: rgb(149, 97, 226);"
    >
      dolor 
    </span>
    <span
      class="px-1 py-0.5 hover:bg-gray-100 hover:cursor-pointer"
      style="color: rgb(246, 109, 155);"
    >
      sit 
    </span>
    <span
      class="px-1 py-0.5 hover:bg-gray-100 hover:cursor-pointer"
      style="color: rgb(246, 153, 63);"
    >
      amet, 
    </span>
    <span
      class="px-1 py-0.5 hover:bg-gray-100 hover:cursor-pointer"
      style="color: rgb(56, 193, 114);"
    >
      consectetur 
    </span>
    <span
      class="px-1 py-0.5 hover:bg-gray-100 hover:cursor-pointer"
      style="color: rgb(77, 192, 181);"
    >
      adipiscing 
    </span>
    <span
      class="px-1 py-0.5 hover:bg-gray-100 hover:cursor-pointer"
      style="color: rgb(108, 178, 235);"
    >
      elit, 
    </span>
    <span
      class="px-1 py-0.5 hover:bg-gray-100 hover:cursor-pointer"
      style="color: rgb(52, 144, 220);"
    >
      sed 
    </span>
  </div>
</div>
`;
